src/pages/Models/Details/ModelDetails.js

```diff
@@ -269,51 +200,44 @@ const ModelDetails = () => {
   const handleAppRowClick = (e, app) => {
     const currentApp = cloneDeep(app);
     currentApp.name = e.currentTarget.dataset.app;
-    setSlidePanelData({ currentApp });
     setFilterByApp(currentApp.name);
+    setPanelType("apps");
+    setEntity(currentApp.name);
+  };
+
+  const closePanel = () => {
+    history.push(`/models/${modelName}`);
+    setPanelType(null);
+    setEntity(null);
   };
 
   const applicationTableRows = useMemo(
     () =>
       generateApplicationRows(modelStatusData, handleAppRowClick, baseAppURL),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
     [baseAppURL, modelStatusData]
   );
 
   const unitTableRows = useMemo(
     () => generateUnitRows(modelStatusData, baseAppURL),
     [baseAppURL, modelStatusData]
   );
-  const unitSlidePanelRows = useMemo(
-    () => generateUnitRows(filteredModelStatusData, baseAppURL),
-    [baseAppURL, filteredModelStatusData]
-  );
 
   const machinesTableRows = useMemo(
     () => generateMachineRows(modelStatusData),
     [modelStatusData]
   );
-  const machinesSlidePanelRows = useMemo(
-    () => generateMachineRows(filteredModelStatusData),
-    [filteredModelStatusData]
-  );
 
   const relationTableRows = useMemo(
     () => generateRelationRows(modelStatusData, baseAppURL),
     [modelStatusData, baseAppURL]
   );
-  const relationSlidePanelRows = useMemo(
-    () => generateRelationRows(filteredModelStatusData, baseAppURL),
-    [filteredModelStatusData, baseAppURL]
-  );
-
-  const appSlidePanelHeader = useMemo(
-    () => generateAppSlidePanelHeader(slidePanelData.currentApp, baseAppURL),
-    [slidePanelData.currentApp, baseAppURL]
-  );
 
   const slidePanelActive = Object.entries(slidePanelData).length > 0;
 
   const [activeView, setActiveView] = useQueryString("view", "status");
+  const [panelType, setPanelType] = useQueryString("panel");
+  const [entity, setEntity] = useQueryString("entity");
```

Comment:
Ahh here is the problem that's causing the QA failure, it's creating two history entries, one for each of these two values. We'll have to use a single query string or update `useQueryString` to be able to support multiple params at once. 

Lets chat about this one.